specs-go/chain/chainid.go

```diff
@@ -0,0 +1,46 @@
+package chain
+
+import "github.com/docker/distribution/digest"
+
+// ChainID takes a slice of digests and returns the ChainID corresponding to
+// the last entry. Typically, these are a list of layer DiffIDs, with the
```

Comment:
> > And the ChainID being returned here corresponds to the whole slice, not just the last entry.
>
> These are the same thing.

No they aren't, and you pointed this out in #482.  If the chain ID was a property of an *entry*, the function signature here would be:

    func ChainID(digest digest.Digest) digest.Digest

and you'd call it with `ChainID(config.RootFS.DiffIDs[3])`, etc.  But the chain ID is really a property of the *slice* (defined for slices with length > 0), so the signature is:

    func ChainID(digests []digest.Digest) (digest.Digest, error)

And you call it with `ChainID(config.RootFS.DiffIDs[:4])`, etc.